assets/js/modules/thank-with-google/components/common/PostTypesSelect.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Chip, ChipCheckmark } from '@material/react-chips';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { ENTER } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { MODULES_THANK_WITH_GOOGLE } from '../../datastore/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function PostTypesSelect() {
+	const postTypes = useSelect( ( select ) =>
+		select( CORE_SITE ).getPostTypes()
+	);
+
+	const buttonPostTypes = useSelect( ( select ) =>
+		select( MODULES_THANK_WITH_GOOGLE ).getButtonPostTypes()
+	);
+
+	const { setButtonPostTypes } = useDispatch( MODULES_THANK_WITH_GOOGLE );
+	const toggleChip = useCallback(
+		( { type, target, keyCode } ) => {
+			if ( type === 'keyup' && keyCode !== ENTER ) {
+				return;
+			}
+
+			const chip = target.closest( '.mdc-chip' );
+			const chipID = chip?.dataset?.chipId; // eslint-disable-line sitekit/acronym-case
+			if ( ! chipID ) {
+				return;
+			}
+
+			if ( buttonPostTypes?.includes( chipID ) ) {
+				setButtonPostTypes(
+					buttonPostTypes.filter(
+						( postType ) => postType !== chipID
+					)
+				);
+			} else {
+				setButtonPostTypes( [ ...( buttonPostTypes || [] ), chipID ] );
+			}
+		},
+		[ buttonPostTypes, setButtonPostTypes ]
+	);
+
+	const options = postTypes?.map( ( { slug, label } ) => (
+		<Chip
+			chipCheckmark={ <ChipCheckmark /> }
+			data-chip-id={ slug }
+			id={ slug }
+			key={ slug }
+			label={ label }
+			onClick={ toggleChip }
+			onKeyUp={ toggleChip }
+			selected={ buttonPostTypes?.includes( slug ) }
+			className="googlesitekit-twg-post-type-select__chip"
+		/>
```

Comment:
I guess this is how the component is supposed to behave, but I find the transition between checked and unchecked a bit jarring, eg:

![CleanShot 2022-07-27 at 20 19 05](https://user-images.githubusercontent.com/90871/181354559-1493413c-d440-4453-aee9-e8d1dc1172f2.gif)

Maybe worth a later discussion as it seems that's how the actual component behaves and it's not something special we're doing. 🤷🏻‍♂️

Reply:
Yeah, probably worth creating a new ticket if we want to discuss if further.